### PR TITLE
adds a switch to include the error information or not

### PR DESCRIPTION
**Pull request by baywet**
related https://github.com/microsoftgraph/msgraph-metadata/pull/110

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoftgraph/MSGraph-SDK-Code-Generator/pull/695)

### CODE REVIEW

File: scripts/run-typewriter-clean-metadata.ps1 [hunk 1]
```diff
@@ -7,13 +7,21 @@ Push-Location -Path $env:TypewriterDirectory
...
-    & $env:TypewriterExecutable -v Info -m $env:InputMetadataFile -o $env:OutputPath -g $env:GenerationMode -t $env:Transform -d $env:DocsDirectory -e $env:endpointVersion
+    $additionalFlags += " -a $env:AddInnerErrorDescription"
 }
 
+if($additionalFlags -ne "")
+{
+    $additionalFlags += " -f $env:OutputMetadataFileName"
+}
+
+Invoke-Expression "$env:TypewriterExecutable -v Info -m $env:InputMetadataFile -o $env:OutputPath -g $env:GenerationMode -t $env:Transform -d $env:DocsDirectory -e $env:endpointVersion"
```
Comment (andrueastman) [inline scripts/run-typewriter-clean-metadata.ps1:25]: I believe you still need to pass `$additionalFlags` to this command.

Comment (baywet) [reply scripts/run-typewriter-clean-metadata.ps1:25]:
```suggestion
Invoke-Expression "$env:TypewriterExecutable -v Info -m $env:InputMetadataFile -o $env:OutputPath -g $env:GenerationMode -t $env:Transform -d $env:DocsDirectory -e $env:endpointVersion$additionalFlags"
```
thanks for catching that. I haven't had the opportunity to test the batch of PRs in this and the metadata repo as I'm waiting for Hidi to release first.

Comment (andrueastman) [reply scripts/run-typewriter-clean-metadata.ps1]: Just to confirm, will the expression still be evaluated without a space between the two variables?
`$env:endpointVersion$additionalFlags` vs `$env:endpointVersion $additionalFlags`

Comment (baywet) [reply scripts/run-typewriter-clean-metadata.ps1]: yep, that's expected, as you can see additionalFlags starts with a blank space, so we don't have to have custom logic here

**Review by andrueastman — APPROVED** [review]

**Review by zengin — DISMISSED** [review]

File: scripts/run-typewriter-clean-metadata.ps1 [hunk 2]
```diff
@@ -7,13 +7,21 @@ Push-Location -Path $env:TypewriterDirectory
...
-    & $env:TypewriterExecutable -v Info -m $env:InputMetadataFile -o $env:OutputPath -g $env:GenerationMode -t $env:Transform -d $env:DocsDirectory -e $env:endpointVersion
+    $additionalFlags += " -a $env:AddInnerErrorDescription"
 }
 
+if($additionalFlags -ne "")
+{
+    $additionalFlags += " -f $env:OutputMetadataFileName"
+}
+
+Invoke-Expression "$env:TypewriterExecutable -v Info -m $env:InputMetadataFile -o $env:OutputPath -g $env:GenerationMode -t $env:Transform -d $env:DocsDirectory -e $env:endpointVersion$additionalFlags"
```
Comment (baywet) [inline scripts/run-typewriter-clean-metadata.ps1:25]:
```suggestion
& $env:TypewriterExecutable -v Info -m $env:InputMetadataFile -o $env:OutputPath -g $env:GenerationMode -t $env:Transform -d $env:DocsDirectory -e $env:endpointVersion$additionalFlags
```

**Pull request merged**